controllers/cluster_controller.go

```diff
@@ -429,6 +429,12 @@ func splitMachineList(list *clusterv1.MachineList) (*clusterv1.MachineList, *clu
 func (r *ClusterReconciler) reconcileControlPlaneInitialized(ctx context.Context, cluster *clusterv1.Cluster) error {
 	logger := r.Log.WithValues("cluster", cluster.Name, "namespace", cluster.Namespace)
 
+	// Do not reconcile the ControlPlaneInitialized if using a ControlPlane provider, that is done
```

Comment:
Nit: What do you think about something like "Skip checking if the control plane is initialized when using a control plane provider"?

Reply:
Done!